packages/stark-ui/src/modules/generic-search/classes/abstract-search-component.ts

```diff
@@ -26,12 +26,12 @@ export abstract class AbstractStarkSearchComponent<SearchResultsType, CriteriaTy
 	 * @internal
 	 * @ignore
 	 */
-	private _latestResults: Readonly<SearchResultsType[]>;
+	private _latestResults: Readonly<SearchResultsType[]> = [];
```

Comment:
I don't think that setting an empty array by default has the same effect because then the developer won't have a way to know whether a search has been triggered or not (the results will always be an array). I think we can use the `!` modifier instead because we know that at some point there will be results.

Reply:
I looked at the code a bit better and I actually think an empty array is more correct here, or making it optional. Because it will only contain a value if `preserveLatestResults` is set to true

Reply:
True, I think we should make it optional (`?`) then ;)